Remove WRITE_EXTERNAL_STORAGE permission check for internal storage access

mkremer-jemmic opened the pull request:
This pull request removes the need of the _WRITE_EXTERNAL_STORAGE_ permission for most of this module's methods on Android. As most of the functions only work with the app's internal storage, they do not need any additional permission. This addresses the issue #905.

Overview of the changes:
- The _clean_ and _openPicker_ methods does no longer check for the _WRITE_EXTERNAL_STORAGE_ permission as it only clears a folder from the internal cache
- The _openCamera_ method does no longer require the _WRITE_EXTERNAL_STORAGE_ permission. If this permission is granted, photos and videos taken will be stored to the external storage as they have been until now. Otherwise, they will be temporarily stored in the app's internal cache storage (optional behavior).

danidfm commented:
Yes please, this is an unnecessary permissions that could affect acceptance from the users. This PR is mandatory.

ivpusic commented:
when running example application and trying to select image, I get permission deined. could you take a look into it?

mkremer-jemmic commented:
I noticed this too, although it worked at one point. I will have a look into this

mkremer-jemmic commented:
The error occurred because the Compression class tried to save (non-)resized images to an external path. I fixed this by checking whether or not the WRITE_EXTERNAL_STORAGE permission has been granted. It follows the same logic as in my first commit for this PR.
Also, I added the cache path to the file provider paths (provider_paths.xml) as taking pictures with the camera did otherwise not work for me in the example project.

gabgagnon commented:
@mkremer-jemmic So with your PR, we no longer upload the file on the External Storage, right?

osrl commented:
Hello, I think even READ_EXTERNAL_STORAGE is not needed to request media from gallery app. Media provider(gallery app) already gives URI based permission for only the requested medias. Please watch this talk from Ian Lake https://youtu.be/C28pvd2plBA?t=1472
